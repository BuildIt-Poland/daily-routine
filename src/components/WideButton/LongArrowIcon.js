import React from 'react';
import styled from 'styled-components';

import { LongArrow } from '../Icons';

const Wrapper = styled.span`
  position: absolute;
  right: 1.3rem;
  bottom: 1.3rem;
`;

function LongArrowIcon() {
  return (
    <Wrapper>
      <LongArrow />
    </Wrapper>
  );
}

export default LongArrowIcon;
