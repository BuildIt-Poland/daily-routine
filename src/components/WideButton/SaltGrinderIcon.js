import React from 'react';
import styled from 'styled-components';

import { SaltGrinder } from '../Icons';

const Wrapper = styled.span`
  position: absolute;
  right: 0.4rem;
  bottom: -0.9rem;
`;

function SaltGrinderIcon() {
  return (
    <Wrapper>
      <SaltGrinder />
    </Wrapper>
  );
}

export default SaltGrinderIcon;
