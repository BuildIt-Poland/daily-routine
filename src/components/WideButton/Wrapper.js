import styled from 'styled-components';

import { colorWhite, fontLarge, fontWeightBold, borderRadius, spacingMedium } from '../../styles/designTokens';

const Wrapper = styled.button`
  position: relative;
  background: ${({ colourset }) => colourset.secondaryColor};
  color: ${colorWhite};
  display: flex;
  align-items: center;
  padding: 0 ${spacingMedium};
  margin: ${spacingMedium} 0;
  border: 0;
  border-bottom: 0.3rem solid ${({ colourset }) => colourset.tertiaryColor};
  border-radius: ${borderRadius};
  font-size: ${fontLarge};
  font-weight: ${fontWeightBold};
  text-align: left;
  text-decoration: none;
  width: 100%;
  height: 4.7rem;

  &:not(:disabled) {
    cursor: pointer;
  }
`;

export default Wrapper;
