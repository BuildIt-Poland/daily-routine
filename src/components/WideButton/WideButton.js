import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import generateColors from '../../utils/generateColors';
import {
  BRAG_VARIANT,
  CONFESS_VARIANT,
  FRONT_END_VARIANT,
  BACK_END_VARIANT,
  DEV_OPS_VARIANT
} from '../../constants/coloursVariants';
import Wrapper from './Wrapper';
import LightbublIcon from './LightbulbIcon';
import SaltGrignderIcon from './SaltGrinderIcon';
import LongArrowIcon from './LongArrowIcon';

const ICONS = {
  [BRAG_VARIANT]: LightbublIcon,
  [CONFESS_VARIANT]: SaltGrignderIcon,
  [FRONT_END_VARIANT]: LongArrowIcon,
  [BACK_END_VARIANT]: LongArrowIcon,
  [DEV_OPS_VARIANT]: LongArrowIcon
};

function WideButton({ variant = 'frontend', children, to, onClick }) {
  // Render link if 'to' prop is passed, button with handler otherwise
  const props = {
    ...(to ? { as: Link, to } : { onClick: onClick })
  };

  const Icon = ICONS[variant];

  return (
    <Wrapper colours={generateColors(variant)} {...props}>
      {children}
      {<Icon />}
    </Wrapper>
  );
}

WideButton.propTypes = {
  children: PropTypes.node,
  to: PropTypes.string,
  onClick: PropTypes.func,
  variant: PropTypes.string
};

export default WideButton;
