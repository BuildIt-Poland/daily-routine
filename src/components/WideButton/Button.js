import React from 'react';
import PropTypes from 'prop-types';

import StyledButton from './StyledButton';

function Button({ children, icon, className }) {
  return (
    <StyledButton className={className}>
      {children}
      {icon}
    </StyledButton>
  );
}

Button.propTypes = {
  children: PropTypes.node,
  icon: PropTypes.element,
  className: PropTypes.string
};

export default Button;
