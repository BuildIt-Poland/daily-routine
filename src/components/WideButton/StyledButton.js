import styled from 'styled-components';

import Button from '../Button';
import {
  colorWhite,
  fontLarge,
  fontWeightBold,
  borderRadius,
  spacingMedium,
  darkGreen,
  colorGreen
} from '../../styles/designTokens';

const StyledButton = styled(Button)`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: ${colorGreen};
  color: ${colorWhite};
  width: 100%;
  height: 4.7rem;
  margin: ${spacingMedium} 0;
  padding: 0 ${spacingMedium};
  border: 0;
  border-bottom: 0.3rem solid ${darkGreen};
  border-radius: ${borderRadius};
  font-size: ${fontLarge};
  font-weight: ${fontWeightBold};
  text-align: left;

  &:not(:disabled) {
    cursor: pointer;
  }
`;

export default StyledButton;
