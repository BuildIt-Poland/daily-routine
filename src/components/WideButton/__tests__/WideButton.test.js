import React from 'react';
import { create } from 'react-test-renderer';

import WideButton from '../WideButton';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

describe('COMPONENT - BragButton', () => {
  it('renders Link if to prop is provided', () => {
    const component = create(<WideButton to="/frontend" />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('render button if onClick is provided', () => {
    const component = create(<WideButton onClick={jest.fn()} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
