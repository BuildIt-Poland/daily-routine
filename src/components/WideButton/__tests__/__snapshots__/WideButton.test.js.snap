// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - BragButton render button if onClick is provided 1`] = `
.c0 {
  position: relative;
  background: #f5ca1d;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 2rem;
  margin: 2rem 0;
  border: 0;
  border-bottom: 0.3rem solid #d18b13;
  border-radius: 0.6rem;
  font-size: 1.8rem;
  font-weight: 700;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  height: 4.7rem;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c1 {
  position: absolute;
  right: 1.3rem;
  bottom: 1.3rem;
}

<button
  className="c0"
  onClick={[MockFunction]}
>
  <span
    className="c1"
  >
    <svg
      height="17"
      viewBox="0 0 28 17"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19.5.015l-2.121 2.121L22.243 7H0v3h22.243l-4.864 4.864 2.121 2.121L27.985 8.5z"
        fill="#fff"
      />
    </svg>
  </span>
</button>
`;

exports[`COMPONENT - BragButton renders Link if to prop is provided 1`] = `
.c0 {
  position: relative;
  background: #f5ca1d;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 2rem;
  margin: 2rem 0;
  border: 0;
  border-bottom: 0.3rem solid #d18b13;
  border-radius: 0.6rem;
  font-size: 1.8rem;
  font-weight: 700;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  height: 4.7rem;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c1 {
  position: absolute;
  right: 1.3rem;
  bottom: 1.3rem;
}

<Link
  className="c0"
  colourset={
    Object {
      "primaryColor": "#f2de6f",
      "secondaryColor": "#f5ca1d",
      "tertiaryColor": "#d18b13",
    }
  }
  to="/frontend"
>
  <span
    className="c1"
  >
    <svg
      height="17"
      viewBox="0 0 28 17"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19.5.015l-2.121 2.121L22.243 7H0v3h22.243l-4.864 4.864 2.121 2.121L27.985 8.5z"
        fill="#fff"
      />
    </svg>
  </span>
</Link>
`;

exports[`COMPONENT - WideButton render button if onClick is provided 1`] = `
.c0 {
  position: relative;
  background: #f5ca1d;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 2rem;
  margin: 2rem 0;
  border: 0;
  border-bottom: 0.3rem solid #d18b13;
  border-radius: 0.6rem;
  font-size: 1.8rem;
  font-weight: 700;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  height: 4.7rem;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c1 {
  position: absolute;
  right: 1.3rem;
  bottom: 1.3rem;
}

<button
  className="c0"
  onClick={[MockFunction]}
>
  <span
    className="c1"
  >
    <svg
      height="17"
      viewBox="0 0 28 17"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19.5.015l-2.121 2.121L22.243 7H0v3h22.243l-4.864 4.864 2.121 2.121L27.985 8.5z"
        fill="#fff"
      />
    </svg>
  </span>
</button>
`;

exports[`COMPONENT - WideButton renders Link if to prop is provided 1`] = `
.c0 {
  position: relative;
  background: #f5ca1d;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 2rem;
  margin: 2rem 0;
  border: 0;
  border-bottom: 0.3rem solid #d18b13;
  border-radius: 0.6rem;
  font-size: 1.8rem;
  font-weight: 700;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  height: 4.7rem;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c1 {
  position: absolute;
  right: 1.3rem;
  bottom: 1.3rem;
}

<Link
  className="c0"
  colourset={
    Object {
      "primaryColor": "#f2de6f",
      "secondaryColor": "#f5ca1d",
      "tertiaryColor": "#d18b13",
    }
  }
  to="/frontend"
>
  <span
    className="c1"
  >
    <svg
      height="17"
      viewBox="0 0 28 17"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19.5.015l-2.121 2.121L22.243 7H0v3h22.243l-4.864 4.864 2.121 2.121L27.985 8.5z"
        fill="#fff"
      />
    </svg>
  </span>
</Link>
`;
